google-cloud-spanner/src/test/java/com/google/cloud/spanner/it/ITTransactionTest.java

```diff
@@ -630,4 +631,23 @@ public Long run(TransactionContext transaction) throws Exception {
       assertThat(e.getErrorCode()).isEqualTo(ErrorCode.INVALID_ARGUMENT);
     }
   }
+
+  @Test
+  public void transactionRunnerReturnsCommitStats() {
+    assumeFalse("Emulator does not return commit statistics", isUsingEmulator());
```

Comment:
The assume here is surprising. Why would this be sometimes be true and sometimes not true in this one test method?

Reply:
`isUsingEmulator()` checks whether the environment variable `SPANNER_EMULATOR_HOST` has been set to a non-empty value. If so, the integration tests are running against the emulator. The emulator does not support all features of Cloud Spanner, which means we need to skip some specific tests. The CI environments runs the tests both against the emulator and Cloud Spanner.